python/cuml/decomposition/pca.pyx

```diff
@@ -736,3 +739,9 @@ class PCA(Base):
             'X_types_gpu': ['2darray', 'sparse'],
             'X_types': ['2darray', 'sparse']
         }
+
+    def _check_is_fitted(self, attr):
```

Comment:
The function looks good for the specific case of PCA, but it seems to me that this is something we can generalize and automate for all estimator `predict/transform/predict_proba` etc, by using estimator tags (in particular `requires_fit`) and standardizing the behavior of initialization of our estimators. Maybe a follow up PR? Could you open a github issue?

Reply:
Very good idea, I can work on generalizing the `NotFittedError` to all the modeles with this PR. I will work on it with the estimator tags.

Reply:
I added an issue #3278 for this task and I will do an additionnal PR.